packages/coinstac-ui/app/render/components/consortium-single.js

```diff
@@ -133,11 +134,44 @@ class ConsortiumSingle extends Component {
               key: fieldIndex,
               label,
               onChange: null,
-              options: values,
               type,
               value: null,
             };
 
+            if (type === 'covariates') {
+              /**
+               * Handle covariate field changes.
+               *
+               * @todo Move this to a reducer
+               *
+               * @param {(Object|null)} changeValue
+               * @param {number} itemsIndex
+               */
+              fieldProps.onChange = (changeValue, itemsIndex) => {
```

Comment:
This mutates the component’s internal state. Should be moved to a reducer. We could also adopt redux-form for this component.